### ISSUE TITLE
Using `st_point` type in migration writes out a schema that cannot be reloaded.

### ISSUE THREAD

**Issue by andrewhao** [rgeo/activerecord-postgis-adapter#161]
I'm unclear to the internals of this adapter, but I think something may be off here:

In a migration:

```ruby
class CreateTrackPoints < ActiveRecord::Migration
  def change
    create_table :track_points do |t|
      t.st_point :coordinate, geographic: true
    end
  end
end
```

Executing this writes out this into my `schema.rb`:

```ruby
create_table "track_points", force: :cascade do |t|
  t.geography "coordinate", limit: {:srid=>4326, :type=>"point", :geographic=>true}
end
```

That's fine and well, but any time I attempt to run any rake command involving `db:schema:load`, I get the following error:

```
/opt/rubies/2.2.0/bin/ruby -I/opt/rubies/2.2.0/lib/ruby/gems/2.2.0/gems/rspec-core-3.1.7/lib:/opt/rubies/2.2.0/lib/ruby/gems/2.2.0/gems/rspec-support-3.1.2/lib /opt/rubies/2.2.0/
lib/ruby/gems/2.2.0/gems/rspec-core-3.1.7/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb  --exclude-pattern spec/features/\*\*/\*_spec.rb
rake aborted!
ActiveRecord::StatementInvalid: PG::InvalidParameterValue: ERROR:  Invalid geometry type modifier: geography
LINE 1: ...ck_points" ("id" serial primary key, "coordinate" geometry(G...
                                                             ^
: CREATE TABLE "track_points" ("id" serial primary key, "coordinate" geometry(GEOGRAPHY,4326), "elevation" decimal, "heart_rate" integer, "time" timestamp, "track_id" integer)
/Users/andrewhao/workspace/velocitas-core/db/schema.rb:35:in `block in <top (required)>'
/Users/andrewhao/workspace/velocitas-core/db/schema.rb:14:in `<top (required)>'
-e:1:in `<main>'
PG::InvalidParameterValue: ERROR:  Invalid geometry type modifier: geography
LINE 1: ...ck_points" ("id" serial primary key, "coordinate" geometry(G...
                                                             ^
/Users/andrewhao/workspace/velocitas-core/db/schema.rb:35:in `block in <top (required)>'
/Users/andrewhao/workspace/velocitas-core/db/schema.rb:14:in `<top (required)>'
-e:1:in `<main>'
Tasks: TOP => db:test:load_schema
(See full trace by running task with --trace)
```

Would it make sense to have the adapter write out the following statement instead into schema.rb?

```diff
+    t.st_point "coordinate", :geographic=>true
-    t.geography "coordinate", limit: {:srid=>4326, :type=>"point", :geographic=>true}
```

Doing so seems to fix the problem.

**Comment by mjy** [comment]
We've replicated the problem.

**Comment by pcai** [comment]
This is a horrible hack (I'm not sure where this logic belongs really), but this monkeypatch seems to work when placed in initializers:


    require 'activerecord-postgis-adapter'
    module ActiveRecord
      module ConnectionAdapters
        module PostGISAdapter
          module OID
            class Spatial < Type::Value
              # FIXME terrible hack.
              # https://github.com/rgeo/activerecord-postgis-adapter/issues/161
              def type
                if @geo_type == "Polygon"
                  return :st_polygon
                elsif @geo_type == "Point"
                  return :st_point
                end
                geographic? ? :geography : :geometry
              end
              
            end
          end
        end
      end
    end

At this point I don't know if this 100% works, but it gets all our tests passing, in case this helps someone.

**Comment by afn** [comment]
Let me know if the fix in #166 works for you.

```ruby
gem 'activerecord-postgis-adapter', github: 'amitree/activerecord-postgis-adapter'
```
